appinventor/components/src/com/google/appinventor/components/runtime/Chart.java

```diff
@@ -0,0 +1,276 @@
+            defaultValue = Component.CHART_TYPE_LINE + "")
+    @SimpleProperty(description = "Specifies the chart's type (line, scatter," +
+            "area, bar, pie).",
+            userVisible = false)
+    public void Type(int type) {
+        this.type = type;
+
+        // Keep track whether a ChartView already exists,
+        // in which case it will have to be reinitialized.
+        boolean chartViewExists = (chartView != null);
+
+        // ChartView currently exists in root layout. Remove it.
+        if (chartViewExists) {
+            view.removeView(chartView.getView());
+        }
+
+        // Create a new Chart view based on the supplied type
+        chartView = createChartViewFromType(type);
+
+        // Add the new Chart view as the first child of the root RelativeLayout
+        view.addView(chartView.getView(), 0);
+
+        // If a ChartView already existed, then the Chart
+        // has to be reinitialized.
+        if (chartViewExists) {
+            reinitializeChart();
+        }
+    }
+
+    /**
+     * Creates and returns a ChartView object based on the type
+     * (integer) provided.
+     * @param type one of {@link Component#CHART_TYPE_LINE},
+     *  {@link Component#CHART_TYPE_SCATTER},
+     *  {@link Component#CHART_TYPE_AREA},
+     *  {@link Component#CHART_TYPE_BAR} or
+     *  {@link Component#CHART_TYPE_PIE}
+     * @return  new ChartView instance
+     */
+    private ChartView createChartViewFromType(int type) {
+        switch(type) {
+            case 0:
+                // Line Chart
+                return new LineChartView(container.$context());
+            case 1:
+                // Scatter Chart
+                return new LineChartView(container.$context());
+            case 2:
+                // Area Chart
+                return new LineChartView(container.$context());
+            case 3:
+                // Bar Chart
+                return new LineChartView(container.$context());
+            case 4:
+                // Pie Chart
+                return new LineChartView(container.$context());
+            default:
+                // Invalid argument
+                throw new IllegalArgumentException("type:" + type);
+        }
+    }
+
+    /**
+     * Reinitializes the Chart view by reattaching all the Data
+     * components and setting back all the properties.
+     */
+    private void reinitializeChart() {
+        // Iterate through all attached Data Components and reinitialize them.
+        // This is needed since the Type property is registered only after all
+        // the Data components are attached to the Chart.
+        // This has no effect when the Type property is default (0), since
+        // the Data components are not attached yet, making the List empty.
+        for (ChartDataBase dataComponent : dataComponents) {
+            dataComponent.initChartData();
+        }
+
+        Description(description);
+        BackgroundColor(backgroundColor);
```

Comment:
I have not found a way to access all the properties at once, and reset them in the Android components.

The reason the properties have to be re-set is because the Type property only gets set only after most of the other properties have already been set, so they have to be applied to the new Chart view.

Any suggestions for this approach?